### PR TITLE
NEW Add silverstripe/login-forms

### PR DESCRIPTION
**Pull request by chillu**
See https://github.com/silverstripe/recipe-cms/issues/26

### CODE REVIEW

**Comment by ScopeyNZ** [comment]
Do we want to wait for a stable release? Probably won't be too far away now - at least before the next recipe release.

**Comment by chillu** [comment]
> Do we want to wait for a stable release?

See https://github.com/silverstripe/recipe-cms/issues/26

File: composer.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
         "php": ">=7.1.0",
         "silverstripe/recipe-plugin": "^1.2",
         "silverstripe/recipe-cms": "4.5.x-dev",
-        "silverstripe-themes/simple": "~3.2.0"
+        "silverstripe-themes/simple": "~3.2.0",
+        "silverstripe/login-forms": "^4.0@beta"
```
Comment (robbieaverill) [inline composer.json]: Perhaps if it's going to be part of the installer recipe we should give it a loose constraint, which cow would adjust as it does with the other recipes already - e.g. `4.0.x-dev`

Comment (chillu) [reply composer.json]: So that'll trigger `cow` to use the latest available tag upon recipe releases, incl. `beta` tags? That'd work.

Comment (robbieaverill) [reply composer.json]: You can set modules to be "upgrade-only" which will prevent cow from trying to tag them during the release process if you want. In that case you'd probably want to use a constraint such as the one you've got here already. You'd probably use `4.0.x-dev` if you're intending to tag login-forms releases during core releases via cow as well.

I agree with @ScopeyNZ that tagging login-forms as stable now seems like an acceptable thing to do.

Comment (chillu) [reply composer.json]: OK, so if I'm understanding this correctly:
- Option A: Tag login-forms stable now, change constraint to `4.0.x-dev` which then triggers cow to pull in the stable release when we tag `4.5.0` recipe
- Option B: Leave login-forms in beta now, change constraint to `4.0.x-dev` and have cow pull in latest beta release when we tag `4.5.0` recipe.

I'm fine with either option.

Comment (ScopeyNZ) [reply composer.json]: (And either of them require the same change here 🙂)

Comment (chillu) [inline composer.json]:
```suggestion
        "silverstripe/login-forms": "^4.0.x-dev"
```

Comment (chillu) [reply composer.json]: Alright, changed to `4.0.x-dev`

File: composer.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
         "php": ">=7.1.0",
         "silverstripe/recipe-plugin": "^1.2",
         "silverstripe/recipe-cms": "4.5.x-dev",
-        "silverstripe-themes/simple": "~3.2.0"
+        "silverstripe-themes/simple": "~3.2.0",
+        "silverstripe/login-forms": "^4.0.x-dev"
```
Comment (robbieaverill) [inline composer.json]:
```suggestion
        "silverstripe/login-forms": "4.0.x-dev"
```

I don't think the previous version works

**Comment by dnsl48** [comment]
This patch didn't make it into `4.5`.
We may need to amend the changelogs.

**Comment by chillu** [comment]
Doh, OK it'll go into the 4.6 release then. I've changed the constraint to `^4.0` since we've got a stable release of this module now.

**Comment by chillu** [comment]
This is ready to merge.

**Pull request merged**